lib/Net/SIP/Simple/Call.pm

```diff
@@ -452,9 +452,17 @@ sub dtmf {
 	    $type = $self->{param}{sdp_peer}
 		&& $self->{param}{sdp_peer}->name2int('telephone-event/8000','audio');
 	} elsif ( $m eq 'audio' ) {
-	    $type = $self->{param}{sdp_peer}
-		&& $self->{param}{sdp_peer}->name2int('PCMU/8000','audio')
-		|| 0; # default id for PCMU/8000
+	    my $sdp_peer = $self->{param}{sdp_peer};
+	    if ($sdp_peer) {
+		$type = $sdp_peer->name2int('PCMU/8000','audio') unless defined $type;
+		$type = $sdp_peer->name2int('PCMA/8000','audio') unless defined $type;
+		if (!defined $type) {
+			my $media = $sdp_peer->get_media();
+			my ($fmt) = map { ($_->{media} eq 'audio') ? $_->{fmt} : () } @$media;
+			($type) = grep { $_ == 0 || $_ == 8 } @$fmt; # Only PCMU=0 and PCMA=8 are supported
```

Comment:
Why not merge these two lines, i.e.

      ($type) =  map { $_->{media} eq 'audio' && ($_->{fmt} == 0 || $_->{fmt} == 8) ? $_->{fmt} : () } @$media;

Reply:
Because $fmt is reference to array. So you cannot write `... eq 'audio' && ($_->{fmt} == ..`. `$_->{fmt}` is arrayref, so above code does not work. You would have to put `grep` into that `map`, something like this:

```perl
($type) = map { ($_->{media} eq 'audio') ? grep { $_ == 0 || $_ == 8 } @{$_->{fmt}} : () } @$media;
```

And I think that map and grep on separate lines is more readable.

But let me know what do you think. I can change it, if you prefer it.